libs/com.etendorx.generate_entities.core/src/main/java/org/etendorx/base/session/OBPropertiesProvider.java

```diff
@@ -0,0 +1,177 @@
+    OBPropertiesProvider.instance = instance;
+  }
+
+  public Properties getOpenbravoProperties() {
+    if (obProperties == null) {
+      readPropertiesFromDevelopmentProject();
+    }
+    return obProperties;
+  }
+
+  /*
+    public Document getFormatXMLDocument() {
+      if (formatXML == null) {
+        final File file = getFileFromDevelopmentPath("Format.xml");
+        if (file != null) {
+          try {
+            SAXReader reader = XMLUtil.getInstance().newSAXReader();
+            formatXML = reader.read(new FileReader(file));
+          } catch (Exception e) {
+            throw new IllegalStateException(e);
+          }
+        }
+      }
+      return formatXML;
+    }
+  */
+  public void setFormatXML(InputStream is) {
+    try {
+      SAXReader reader = XMLUtil.getInstance().newSAXReader();
+      formatXML = reader.read(is);
+    } catch (Exception e) {
+      throw new IllegalStateException(e);
+    }
+  }
+
+  public void setProperties(InputStream is) {
+    if (obProperties != null) {
+      log.debug("Openbravo properties have already been set, setting them again");
+    }
+    log.debug("Setting Openbravo.properties through input stream");
+    obProperties = new Properties();
+    try {
+      obProperties.load(is);
+      is.close();
+
+      if (OBConfigFileProvider.getInstance() == null || OBConfigFileProvider.getInstance()
+          .getServletContext() == null) {
+        log.debug("ServletContext is not set, not trying to override Openbravo.properties");
+        return;
+      }
+
+      ConfigParameters.overrideProperties(obProperties,
+          OBConfigFileProvider.getInstance().getServletContext().getRealPath("/WEB-INF"));
+    } catch (final Exception e) {
+      throw new OBException(e);
+    }
+  }
+
+  public void setProperties(Properties props) {
+    if (obProperties != null) {
+      log.debug("Openbravo properties have already been set, setting them again");
+    }
+    log.debug("Setting openbravo.properties through properties");
+    obProperties = new Properties();
+    obProperties.putAll(props);
+  }
+
+  public void setProperties(String fileLocation) {
+    log.debug("Setting Openbravo.properties through file: " + fileLocation);
+    obProperties = new Properties();
+    try {
+      final FileInputStream fis = new FileInputStream(fileLocation);
+      obProperties.load(fis);
+      fis.close();
+    } catch (final Exception e) {
+      throw new OBException(e);
+    }
+  }
+
+  /**
+   * Looks for a boolean property key and return <code>true</code> in case its value is true or yes
+   * and false other case.
+   */
+  public boolean getBooleanProperty(String key) {
+    Properties properties = getOpenbravoProperties();
+    if (properties == null) {
+      return false;
+    }
+
+    String value = properties.getProperty(key);
+    if (value == null) {
+      return false;
+    }
+
+    return value.equalsIgnoreCase("true") || value.equalsIgnoreCase("yes");
```

Comment:
I make usage of StringUtils instead of BooleanUtils to avoid api changes. The code as is not checking for "Y", "N", "on", "off", etc. BooleanUtils adds risk to get an API change.